### ISSUE TITLE
Unable to splat kwargs when calling function with optional args

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#5255]
**Describe the bug**
When splatting keyword arguments as in `foo(**a_dict)` and if the function parameters differ in their types, a type error is produced that the union of the value types of `a_dict` is incompatible with the union of the parameters types of the function.  This is usual not an issue in practice except when the function takes optional arguments and these are not provided.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
Unsure. Before 1.1.312 this would pass; the new behaviour is more correct but is hampered by the Python typing system's inability to exclude args by name from kwargs.  Perhaps Pyright could interpret a splatted dict literal as a typed dict. 

**Screenshots or Code**
```python
# pyright: strict

def foo(arg: bool = False, **kwargs: str):
    ...

foo(**{'a': 'b'})

```

**VS Code extension or command-line**
1.1.313, either

**Comment by layday** [comment]
I experienced this [here](https://github.com/layday/instawow/blob/ea2754911dd9fed6ed7c89d6530cee099427be52/src/instawow/config.py#L163-L167) when calling https://github.com/python-attrs/cattrs/blob/116d25bd3e00d105ec80b81aa7fc60bcc687f1f4/src/cattrs/gen/__init__.py#L51-L57.  I'm not sure how it'd be able to work around it other than with a `cast`.

**Comment by erictraut** [comment]
Yeah, this change was made in response to [this discussion thread](https://github.com/microsoft/pyright/discussions/5231). There's a tradeoff here, and one could argue for either behavior. One produces false positives, the other produces false negatives. In this case, I opted to be consistent with mypy's behavior. So this is now the expected behavior, and I don't consider it a bug.
